### PR TITLE
Pass Settings to aggregate function creator

### PR DESCRIPTION
**Pull request by vdimir**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by abyss7** [comment]
Can't see in this PR where and how do you use new creator variant, can you show please?

**Comment by vdimir** [comment]
> Can't see in this PR where and how do you use new creator variant, can you show please?

@abyss7 

See commit [`f022f07` (#22762)](https://github.com/ClickHouse/ClickHouse/pull/22762/commits/f022f0744b1f2a6bb80da0c116a9124306efb827). But I'm not intended to make `windowFunnel` experimental (now flag true on default and doesn't make any changes), but I will use it for function from https://github.com/ClickHouse/ClickHouse/pull/19766 after merge.

**Comment by abyss7** [comment]
Please, add a functional test for this setting.

**Comment by vdimir** [comment]
@abyss7 
Added. 
It will be removed from `windowFunnel`, but I'll add same test logic to `sequenceNextNode` when will merge it.

File: src/AggregateFunctions/AggregateFunctionFactory.cpp [hunk 1]
```diff
@@ -137,7 +153,7 @@ AggregateFunctionPtr AggregateFunctionFactory::getImpl(
         if (!out_properties.returns_default_when_only_null && has_null_arguments)
             return nullptr;
 
-        return found.creator(name, argument_types, parameters);
+        return found.create(name, argument_types, parameters, query_context->getSettingsRef());
```
Comment (abyss7) [inline src/AggregateFunctions/AggregateFunctionFactory.cpp:156]: See tests under UBSan: `runtime error: member call on null pointer of type 'DB::Context'`

**Review by abyss7 — CHANGES_REQUESTED** [review]

File: src/AggregateFunctions/AggregateFunctionFactory.cpp (hunk 1 above)
Comment (vdimir) [reply src/AggregateFunctions/AggregateFunctionFactory.cpp]: I changed to passing `Setting *` to creator and add check if query_context initialized
https://github.com/ClickHouse/ClickHouse/pull/22762/files#diff-4a4dba9eabf5c6708dbc69aaf1869432beb57a38415b3ffa62316e78504fff66R141-R142

**Comment by vdimir** [comment]
I removed holding variant of function with Settings or without it and pass Settings to all aggregate function creators.

**Comment by vdimir** [comment]
@abyss7 take a look, please

**Review by abyss7 — APPROVED** [review]

**Pull request merged**